### PR TITLE
Mail

### LINKED ISSUE #1: [Enhancement]Send summary by email
1. Share something
2. The list of links is diplayed on screen
The same result should be shared by email too

### PR DESCRIPTION
**Pull request by GuillaumeRand**
Adding the mail sending function with php

### CODE REVIEW

File: lib/functions.js.php [hunk 1]
```diff
@@ -777,10 +781,89 @@ function copyLinkToClipboard(link_id) {
...
+            {
+                pop_failure (res);
+                return;
+            }
+        } else {
+            ("<?php echo t("ERR_OCC"); ?>");
+        }
+    }
+    req.open ("POST", 'script.php', true);
+    req.withCredentials = true;
```
Comment (GuillaumeRand) [inline lib/functions.js.php:846]: remove req.withCredentials = true;

File: lib/functions.php [hunk 2]
```diff
@@ -1781,3 +1781,104 @@ function jirafeau_add_ending_slash($path)
...
+            align-items: center;
+        }
+        #mail {
+            background-color:#f4f4f4;
+            max-width: 60%;
+            display:flex;
+            flex-direction: column;
+            align-items:center;
+        }
+    </style>
```
Comment (GuillaumeRand) [inline lib/functions.php:1848]: revoir le css

File: lib/functions.php [hunk 3]
```diff
@@ -1781,3 +1781,104 @@ function jirafeau_add_ending_slash($path)
...
+                <p>' . $message .'</p>
+            </div>
+            <button id="button" href=' . $link .'>Get your file</button>
+            <div class="block">
+                <h3 class="h3">Dowload link:</h3>
+                <p>' . $link . '</p>
+                <h3 class="h3">Files: ' . $filename .'</h3>
+                <h4>Password:' . $password . '</h4>
+            </div>
+            <p id="footer">To make sure our emails arrive, please add noreply@send.smart4.io to your contacts.</p>
```
Comment (GuillaumeRand) [inline lib/functions.php]: remove this line

File: index.php [hunk 4]
```diff
@@ -145,6 +145,24 @@
...
+
+    <!--- Form send mail -->
+    <div id="ConteneurFormMail">
+        <form name="FormMail" action="" method="post" id="form">
+            <div id="data">
+                <textarea name="filename" id="filename"></textarea>
+                <textarea name="expireDate" id="expireDate"></textarea>
+                <textarea name="link" id="link"></textarea>
+                <textarea name="password" id="password"></textarea>
+            </div>
```
Comment (GuillaumeRand) [inline index.php]: methode maladroite

File: index.php [hunk 5]
```diff
@@ -145,6 +145,24 @@
...
+            <input name="transmitter" type="text" placeholder="<?php echo t('TRANSMITTER'); ?>" id="transmitter" required/>
+            <input name="recipient" type="text" placeholder="<?php echo t('RECIPIENT'); ?>" required/>
+            <span id="champs_1"><a href="javascript:create_champ(1)">+</a></span>
```
Comment (GuillaumeRand) [inline index.php]: methode maladroite

File: lib/functions.js.php [hunk 6]
```diff
@@ -777,10 +781,89 @@ function copyLinkToClipboard(link_id) {
...
+
+function getPassword() {
+    let password = document.getElementById("input_key").value;
+    document.getElementById('password').textContent = password;
+}
+
+function create_champ(i) {
+let i2 = i + 1;
+document.getElementById('champs_'+i).innerHTML = '<input type="text" name="recipient"></span>';
+document.getElementById('champs_'+i).innerHTML += (i <= 10) ? '<br /><span id="champs_'+i2+'"><a href="javascript:create_champ('+i2+')">+</a></span>' : '';
```
Comment (GuillaumeRand) [inline lib/functions.js.php]: à améliorer, pour éviter de crée un span à chaque fois

File: lib/functions.js.php [hunk 7]
```diff
@@ -764,7 +765,6 @@ function upload_speed_refresh_limiter(speed_str)
 function copyLinkToClipboard(link_id) {
     var focus = document.activeElement;
     var e = document.getElementById(link_id);
-
```
Comment (sebmennetrier) [inline lib/functions.js.php:767]: unnecessary

File: lib/functions.js.php [hunk 8]
```diff
@@ -777,10 +777,91 @@ function copyLinkToClipboard(link_id) {
...
 
 function addCopyListener(button_id, link_id) {
-    if(document.getElementById(button_id)){
-        document.getElementById(button_id)
-            .addEventListener("click", function() {
-                copyLinkToClipboard(link_id);});
+    if(document.getElementById(button_id)) {
+        document.getElementById(button_id).addEventListener("click", function() {
+            copyLinkToClipboard(link_id);
+            let element = document.getElementById(button_id);
```
Comment (sebmennetrier) [inline lib/functions.js.php]: document.getElementById(button_id); should be called only one time

File: lib/functions.js.php [hunk 9]
```diff
@@ -777,10 +777,91 @@ function copyLinkToClipboard(link_id) {
...
+                element.classList.toggle("Copy");
+            } else {
+                let classes = element.className.split(" ");
+                let i = classes.indexOf("Copy");
+                if (i >= 0)
+                    classes.splice(i, 1);
+                else
+                    classes.push("Copy");
+                    element.className = classes.join(" ");
+            }
```
Comment (sebmennetrier) [inline lib/functions.js.php]: I don't see where to remove the class "Copy" from other button :thinking:

File: lib/functions.php [hunk 10]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+            align-items: center;
+        }
+        #mail {
+            background-color:#f4f4f4;
+            max-width: 60%;
+            display:flex;
+            flex-direction: column;
+            align-items:center;
+        }
+    </style>
```
Comment (sebmennetrier) [inline lib/functions.php]: And why do you add id everywhere (#h1 on h1, #body on body). I think that it's very not necessary. Just use css selector body, h1, ... without the #

File: lib/functions.php [hunk 11]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+        #text {
+            margin: 30px;
+        }
+        #button {
+            text-transform: uppercase;
+            outline: 2px;
+            background: #663d1c;
+            width: 35%;
+            padding: 15px;
+            color: #FFFFFF;
```
Comment (sebmennetrier) [inline lib/functions.php]: `color: #fff` is shorter and working

File: lib/functions.php [hunk 12]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+        }
+        .h3 {
+            margin: 0px;
+        }
+        .block {
+            display: flex;
+            flex-direction: column;
+            align-items: center;
+            margin-top:20px;
+            margin-bottom:20px;
```
Comment (sebmennetrier) [inline lib/functions.php]: missing space after `:`

File: lib/functions.php [hunk 13]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+            margin-top: 0px;
+        }
+        .h3 {
+            margin: 0px;
+        }
+        .block {
+            display: flex;
+            flex-direction: column;
+            align-items: center;
+            margin-top:20px;
```
Comment (sebmennetrier) [inline lib/functions.php]: missing space after `:`

File: lib/functions.php [hunk 14]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+ * @param $recipient: the recipient of the email
+ * @param $message is the message of the email
+ * @return true if the mail send.
+ */
+function jirafeau_send_mail($transmitter, $recipient, $message, $link, $email_subject, $filename, $expireDate, $password)
+{
+    $message = '<html>
+    <style>
+        #h1 {
+            color:#663d1c;
```
Comment (sebmennetrier) [inline lib/functions.php]: missing space after `:`

File: lib/functions.php [hunk 15]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+        }
+        #body {
+            justify-content:center;
```
Comment (sebmennetrier) [inline lib/functions.php]: missing space after `:`

File: lib/functions.php [hunk 16]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+            padding: 15px;
+            color: #FFFFFF;
+            font-size: 14px;
+            cursor: pointer;
+            border-radius: 20px;
+        }
+        #footer {
+            font-size: 12px;
+            display: flex;
+            flex-direction:column;
```
Comment (sebmennetrier) [inline lib/functions.php]: missing space after `:`

File: lib/functions.php [hunk 17]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+            align-items: center;
+        }
+        #mail {
+            background-color:#f4f4f4;
```
Comment (sebmennetrier) [inline lib/functions.php]: missing space after `:`

File: lib/functions.php [hunk 18]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+            max-width: 60%;
+            display:flex;
```
Comment (sebmennetrier) [inline lib/functions.php]: missing space after `:`

File: lib/functions.php [hunk 19]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+            flex-direction: column;
+            align-items:center;
```
Comment (sebmennetrier) [inline lib/functions.php]: missing space after `:`

File: lib/functions.php [hunk 20]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+        }
+    </style>
+    <head>
+        <title>Jirafeau</title>
```
Comment (sebmennetrier) [inline lib/functions.php]: Jyrafon

File: lib/functions.php [hunk 21]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+    </head>
+    <body id="body">
+        <div id="mail">
+            <h1 id="h1">Jirafeau</h1>
```
Comment (sebmennetrier) [inline lib/functions.php]: Jyrafon

File: lib/functions.php [hunk 22]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+            <img src="https://framalibre.org/sites/default/files/leslogos/JirafeauLogo.jpeg">
```
Comment (sebmennetrier) [inline lib/functions.php]: oO, now it's Jyrafon, do not use logo at the moment

File: media/courgette/style.css.php [hunk 23]
```diff
@@ -469,3 +469,59 @@
...
+  background: #663d1c;
+  width: 100%;
+  border: 0;
+  padding: 15px;
+  color: #FFFFFF;
+  font-size: 14px;
+  -webkit-transition: all 0.3 ease;
+  transition: all 0.3 ease;
+  cursor: pointer;
+}
```
Comment (sebmennetrier) [inline media/courgette/style.css.php]: Using existing classes do not work ?

File: lib/functions.php [hunk 24]
```diff
@@ -1781,3 +1781,103 @@ function jirafeau_add_ending_slash($path)
...
+    </html>';
+    $headers[] = 'From: '.$transmitter;
+    $headers[] = "Reply-to:" . $transmitter;
+    $headers[] = 'To: '.$recipient;
+    $headers[] = "Importance: Normal";
+    $headers[] = "MIME-Version: 1.0";
+    $headers[] = 'Content-Type: text/html; charset="utf-8"';
+    $headers[] = 'Content-Tranfert-Encoding: 8bit';
+    return mail($recipient, $email_subject, $message, implode("\r\n", $headers));
+}
```
Comment (sebmennetrier) [inline lib/functions.php:1904]: I don't understand how to manage an array of recipient. Do you send one mail per recipient, or one for every recipients (in "To" or in "CC" or in "CCI").

File: lib/functions.js.php (hunk 7 above)
Comment (GuillaumeRand) [reply lib/functions.js.php:767]: I took it off

File: lib/functions.js.php (hunk 8 above)
Comment (GuillaumeRand) [reply lib/functions.js.php]: indeed, I corrected this

File: lib/functions.js.php (hunk 9 above)
Comment (GuillaumeRand) [reply lib/functions.js.php]: it's normal ! I haven't finished it yet

File: lib/functions.php (hunk 14 above)
Comment (GuillaumeRand) [reply lib/functions.php]: it's corrected

File: lib/functions.php (hunk 15 above)
Comment (GuillaumeRand) [reply lib/functions.php]: it's corrected

File: lib/functions.php (hunk 13 above)
Comment (GuillaumeRand) [reply lib/functions.php]: it's corrected

File: lib/functions.php (hunk 12 above)
Comment (GuillaumeRand) [reply lib/functions.php]: it's corrected

File: lib/functions.php (hunk 11 above)
Comment (GuillaumeRand) [reply lib/functions.php]: it's corrected

File: lib/functions.php (hunk 17 above)
Comment (GuillaumeRand) [reply lib/functions.php]: it's corrected

File: lib/functions.php (hunk 10 above)
Comment (GuillaumeRand) [reply lib/functions.php]: I thought emails only accepted styles with id, I was wrong

File: lib/functions.php (hunk 20 above)
Comment (GuillaumeRand) [reply lib/functions.php]: it's corrected

File: lib/functions.php (hunk 21 above)
Comment (GuillaumeRand) [reply lib/functions.php]: it's corrected

File: lib/functions.php (hunk 22 above)
Comment (GuillaumeRand) [reply lib/functions.php]: it's corrected

File: media/courgette/style.css.php (hunk 23 above)
Comment (GuillaumeRand) [reply media/courgette/style.css.php]: I corrected this in part,  not all styles are totally suitable for the form

File: lib/functions.php (hunk 24 above)
Comment (GuillaumeRand) [reply lib/functions.php:1904]: the recipients are retrieved using 
`let NbrRecipients = document.getElementsByName('recipient'); ` , and added to the recipient array in a for :
```
for (element of NbrRecipients) {
        recipient.push (element.value);
    }
```
Then the array is sent with the xhr request, and it is received in script.php as a comma parser string (that also surprised me) sending mail is done for each recipient, they are all in the box "To"
Screenshot_20210212_153411
Screenshot_20210212_153z411
Screenshot_2021-02-12 regtferdbhgtfe - guillaumerand gmail com - Gmail

File: lib/locales/en.json [hunk 25]
```diff
@@ -1,4 +1,10 @@
 {
+    "ALERTSUBMIT": "Send",
+    "ALERTCOPY": "Copy",
```
Comment (sebmennetrier) [inline lib/locales/en.json]: Copy or Copied ?

File: lib/locales/en.json [hunk 26]
```diff
@@ -1,4 +1,10 @@
 {
+    "ALERTSUBMIT": "Send",
```
Comment (sebmennetrier) [inline lib/locales/en.json]: If you really want keep "ALERT" word in your key name, you should use a "_" to separate words like below

**Pull request merged**